File: 2-add-two-numbers/add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
int sum=0;
int carry=0;
ListNode* head=new ListNode(0);
ListNode* a=head;
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        while(l1||l2||carry)
        {   sum=carry;
            if(l1)
            {
                sum+=l1->val;
                l1=l1->next;
            }
            if(l2)
            {
                sum+=l2->val;
                l2=l2->next;
            }
            carry=sum/10;
            a->next=new ListNode(sum%10);
            a=a->next;
            }

         
return head->next;
}};

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        s=0
        place=1
        while (l1):
           s=s+l1.val*place
           place*=10
           l1=l1.next
        s1=0
        place=1
        while (l2):
           s1=s1+l2.val*place
           place*=10
           l2=l2.next
        s3=s1+s
        head=None
        t=None
        if s3==0:
            return ListNode(0)
        while (s3):
            r=s3%10
            s3=s3//10
            newNode=ListNode(r)
            if not head:
                head=newNode
                t=head
                continue
            t.next=newNode
            t=newNode
        return head